tool/resources/org/antlr/v4/tool/templates/codegen/Go/Go.stg

```diff
@@ -658,10 +655,10 @@ switch la_ {
 OptionalBlock(choice, alts, error) ::= <<
 p.SetState(<choice.stateNumber>)
 p.GetErrorHandler().Sync(p)
-la_ := p.GetInterpreter().AdaptivePredict(p.GetTokenStream(), <choice.decision>, p.GetParserRuleContext())
+
 <if(alts)>
 
-<alts:{alt | if la_ == <i><if(!choice.ast.greedy)>+1<endif> {
+<alts:{alt | if p.GetInterpreter().AdaptivePredict(p.GetTokenStream(), <choice.decision>, p.GetParserRuleContext()) == <i><if(!choice.ast.greedy)>+1<endif> {
```

Comment:
@sharwell Not totally sure, but I think you're wrong about that. 

The case shown in #1459 have different parameter sets. In this case, the `decision` id is different in the two invocations.

The code that was previously causing a compiler error is now re-written as:

```
if p.GetInterpreter().AdaptivePredict(p.GetTokenStream(), 3, p.GetParserRuleContext()) == 1 {
		{
			p.SetState(22)
			p.Match(TestParserT__1)
		}

	}
	p.SetState(26)
	p.GetErrorHandler().Sync(p)

	if p.GetInterpreter().AdaptivePredict(p.GetTokenStream(), 4, p.GetParserRuleContext()) == 1 {
		{
			p.SetState(25)
			p.Match(TestParserT__2)
		}

	}
```